### PR TITLE
Issue39 - Readme updates

### PR DESCRIPTION
**Pull request by dcaba**
Given I wanted to update the readme, I also started:
https://github.com/dachad/tcpgoon/issues/39

Still not covering everything (just a TOC generation), but the new README sections and updated media files should be valuable by themselves. The branch was done from the latest commit in the eureka one (which integration, I understand, should not be a problem, despite some code/tests are not exposed), so this is potentially mergeable.

### CODE REVIEW

**Comment by chadell** [comment]
I like your idea @dcaba but you could do it in two phases (if you want), one a static update with the nice Goon Gopher and some new info, and another for the automation.
Regarding the automation, we can run the helper for the Help section (easy), but the Examples will be difficult because flags, commands can change...

**Comment by dcaba** [comment]
My idea is to introduce the first step on the automation here, but extra PRs will be sent to extend the coverage.

Yes, help will be probably the next one. And for the other commands, maybe we can actually connect this to the end-to-end tests (so, if you change a flag, the end-to-end tests should also be updated - maintained on sync, and the autoupdater could pick the commands and output from there...)

**Comment by dcaba** [comment]
Let me know if you would be happy to merge in the current state (after bringing latest master and resolving conflicts we will have because of the upcoming flag changes...)

**Comment by dcaba** [comment]
@chadell , take a look to the final version. In addition to the README updates, the README autoupdater is covering in the end the examples/usage-help, so we don't need to maintain these lines manually anymore, but the associated scripts. I'd say, then, the README autoupdater is done

**Comment by dcaba** [comment]
"final of the final". New section to the README, and a nice godepgraph included dynamically (as part of the readme autoupdater)

File: README.src.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+## TL;DR
+
+Tool to test concurrent connections towards a server listening to a TCP port
+
+## Description
+
+* Given a hostname, port, the number of connections (100 by default), 
+a delay between connections (10ms by default) and an interval between stats
+updates to the standard output...
+* It will use goroutines to open a tcp connection and try to read from it
```
Comment (chadell) [inline README.src.md]: a tcp connection -> tcp connections

File: README.src.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+### Execution using Docker
+
+See our [public docker image and its documentation](https://hub.docker.com/r/dachad/tcpgoon/). The image
+is being updated continuously; you can bind to specific versions, or to the "latest" tag.
+
+
+## Extra project information
+
+### Why do I want to test tcp connections?
```
Comment (chadell) [inline README.src.md]: `tcp` -> `TCP`

File: README.src.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+conditions may not be reproduced during the stress tests execution, but in real life (lots of connections
+queued because of a dependency taking longer to reply than it usually does?)
+ 
+hping is not an actual option for this use case given it won't complete a 3-way handshake,
+so the connection will not reach the accept() syscall of your server/application or fill up your
+tcp backlog.
+
+### Where does the project name come from?
+```
+Goon: /ɡuːn/ noun informal; noun: goon; plural noun: goons ;
```
Comment (chadell) [inline README.src.md]: `/ɡuːn/` -> `/ɡoon/` 😄 (just kidding)

File: README.src.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+### TO-DO
+
+We do use [Github issues](https://github.com/dachad/tcpgoon/issues) to track bugs, improvements and feature requests. Do not hesitate
+to raise new ones, or solve them for us by raising PRs ;)
+
+### Project structure
+
+This project uses a layered topology, where *cmd* (complemented by *cmdutil*) takes care of commands/flags/arguments and uses
+*mtcpclient*, which owns and knows everything about "multiple tcp connections" (including reporting), while *tcpclient*
+only cares about managing single tcp connections. *tcpserver* is just there as a dependency for the other packages' tests.
```
Comment (chadell) [inline README.src.md]: `tcp connections` -> `TCP connection`

File: README.src.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+{godepgraph}
+
+A shared package (*debugging*) is also supplied just as a basic mechanism to control debug output.
+
+### README maintenance
+
+Do not edit README.md, as its autogenerated and your changes will be lost. Consider README.src.md and
```
Comment (chadell) [inline README.src.md]: `autogenerated` -> `autogenerated locally`

**Comment by dcaba** [comment]
@chadell , lack of readability in bash text transformations addressed... does it help?

**Pull request merged**